addons/crm_partner_assign/models/crm_lead.py

```diff
@@ -0,0 +1,103 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import random
+from openerp import api, fields, models, _
+from openerp.exceptions import UserError
+from openerp.addons.base_geolocalize.models.res_partner import geo_find, geo_query_address
+
+
+class CrmLead(models.Model):
+    _inherit = 'crm.lead'
+    partner_latitude = fields.Float(string='Geo Latitude', digits=(16, 5))
+    partner_longitude = fields.Float(string='Geo Longitude', digits=(16, 5))
+    partner_assigned_id = fields.Many2one('res.partner', string='Assigned Partner', track_visibility='onchange', help="Partner this case has been forwarded/assigned to.", index=True)
+    date_assign = fields.Date(string='Assignation Date', help="Last date this case was forwarded/assigned to a partner")
+
+    def get_interested_action(self, interested):
+        try:
+            LeadChannelAct = self.env.ref('crm_partner_assign.crm_lead_channel_interested_act')
+        except ValueError:
+            raise UserError(_("The CRM Channel Interested Action is missing"))
+        action = LeadChannelAct.read()[0]
+        action_context = eval(action['context'])
+        action_context['interested'] = interested
+        action['context'] = str(action_context)
+        return action
+    
+    @api.multi
+    def case_interested(self):
+        self.ensure_one()
+        return self.get_interested_action(True)
+
+    @api.multi
+    def case_disinterested(self):
+        self.ensure_one()
+        return self.get_interested_action(False)
+    
+    @api.multi
+    def assign_salesman_of_assigned_partner(self):
+        for lead in self.filtered(lambda x: x.partner_assigned_id.user_id != x.user_id and x.stage_id.probability > 0 and x.stage_id.probability < 100 or x.stage_id.sequence == 1):
```

Comment:
you have to make appropriate filter. this will be changes in new crm lead case and crm stage.

x.stage_id.probability > 0 and x.stage_id.probability < 100 or x.stage_id.sequence == 1